Revamp the README for the current state of the world (#1594)
hashicorp/terraform-website: README.md

- This repository, under `content/source/docs/`:
    - Terraform Cloud docs
    - Terraform Enterprise docs
    - Extending Terraform
    - Publishing Providers and Modules

    **Notable branches:** `master` is the "live" content that gets deployed to terraform.io. The site gets redeployed for new commits to master.
- [hashicorp/terraform][tf-repo], under `website/docs`:
    - Terraform CLI docs
    - Terraform Language docs

    **Notable branches:** `stable-website` is the "live" content that gets deployed to terraform.io, but docs changes should get merged to master (and/or one of the long-lived version branches) first. See [More About `stable-website`][inpage-stable] below for more details.
- A few remaining provider repos... but those won't be here for long! All but a few have migrated to [the Registry](https://registry.terraform.io), and the rest are leaving soon.

## Deploying Changes to [terraform.io][]

↥ [back to top](#table-of-contents)

- **For changes in this repo:** Merge the PR to master, and the site will automatically deploy in about 20m. 🙌
- **For changes in hashicorp/terraform:** Merge the PR to master. Then, either:

>     - Wait for the next release of the project in question. The changes will be deployed automatically.

Author: Oh yup, good catch. 

```suggestion
    - Wait for the next Terraform release. The changes will be deployed automatically.
```

You don't need to add anything special to a sidebar to get the dynamic JavaScript open/close behavior, but note that the "expand all" and filter controls are only added for sidebars with more than a certain number of links.

> ## Basics of Using Submodules

Reviewer: ```suggestion
## Using Submodules
```

↥ [back to top](#table-of-contents)

> [inpage-submodules]: #basics-of-using-submodules

Reviewer: ```suggestion
[inpage-submodules]: #using-submodules
```

Avoid running `git rm` on a submodule unless you know what you're doing. You usually want `git submodule deinit` instead.

## Finding Broken Links

↥ [back to top](#table-of-contents)

Terraform.io uses a few different link checkers, which run as CircleCI jobs. If a link checking job fails, you can go to the job in CircleCI to find out which link URL caused the problem and which page that link appeared on.

All of these jobs are configured in `./circleci/config.yml`, in this repo and in hashicorp/terraform.

### Global Link Check

We run a global link check for the whole site after every deploy.

- **Where:** The job reports its status in the `#proj-terraform-docs` channel in Hashicorp's Slack.

> - **What:** This job only checks internal links within terraform.io, not external links to the rest of the web. (We deploy a _lot,_ and we don't want to be a nuisance on someone else's servers.)

Reviewer: Will we really deploy that much any more now that we don't deploy every time a provider release happens?

Author: 3 deploys yesterday, 4 on wednesday, 3 on tuesday... so, several times a day. I guess it's a relative question, huh? 

```suggestion
- **What:** This job only checks internal links within terraform.io, not external links to the rest of the web. (It runs frequently, and we don't want to be a nuisance.)
```

- **Who:** The Terraform Education team is ultimately responsible for dealing with any broken links this turns up, but anyone in the channel is welcome to fix something if they see it first!
- **How:** We're using [filiph/linkcheck](https://github.com/filiph/linkcheck/) for this. In addition to checking links, this also warms up the Fastly cache for the site.

### PR Link Check

We run a targeted link check for docs PRs, in this repo and in hashicorp/terraform.

- **Where:** It shows up as a GitHub PR check. It only runs for PRs from people in the HashiCorp GitHub organization (which should be fine, since we're the most likely to change a bunch of links at once.)
- **What:** This job only checks links in the _content area_ (not navs/headers) of _pages that were changed in the current PR._ It checks both internal and external links.
- **Who:** If this job is red in your PR, please fix your broken links before merging! Alternately, if it throws a false-positive and complains about a link that is actually fine, make sure to explain that before merging.

> - **How:** This is a custom-built Ruby script, because we weren't able to find an off-the-shelf link checker that met our requirements (i.e. don't complain about problems that have nothing to do with this PR).

Reviewer: Should we link to the script from here?

Author: ```suggestion
- **How:** This is a custom Ruby script, because we weren't able to find an off-the-shelf link checker that met our requirements (i.e. don't complain about problems that have nothing to do with this PR). ([content/scripts/check-pr-links.rb](./content/scripts/check-pr-links.rb))
```

### Known Incoming Link Check

We run a weekly check to make sure we don't delete popular pages without redirecting them somewhere useful.

- **Where:** The job reports its status mid-morning (PST) every Monday, in the `#proj-terraform-docs` channel in Hashicorp's Slack.
- **What:** This job checks a list of paths from the `content/scripts/testdata/incoming-links.txt` file.
- **Who:** The Terraform Education team is ultimately responsible for dealing with any broken links this turns up, but anyone in the channel is welcome to fix something if they see it first!

> - **How:** This is a custom-built shell script that uses `wget`.

Reviewer: Should we link to it here?

Author: sure!

```suggestion
- **How:** This is a custom shell script that uses `wget`. ([content/scripts/check-incoming-links.sh](https://github.com/hashicorp/terraform-website/blob/master/content/scripts/check-incoming-links.sh))
```
